### PR TITLE
feat(frontend): use libtest-mimic for planner test

### LINKED ISSUE #2881: frontend: use libtest-mimic for planner test
Currently, we are using `build.rs` to generate a list of tests. This doesn't integrate well with most ecosystems. In fact, we can use libtest-mimic to create a test-like interface, and let cargo nextest to run it in parallel.

https://github.com/risinglightdb/risinglight/pull/656

### PR DESCRIPTION
**Pull request by alissa-tung**
Signed-off-by: alissa-tung <alissa-tung@outlook.com>

## What's changed and what's your intention?

use libtest-mimic for planner test, remove `build.rs` which was used for generating planner tests

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

close #2881

### CODE REVIEW

**Comment by alissa-tung** [comment]
cc @skyzh

**Comment by skyzh** [comment]
https://buildkite.com/singularity-data/pull-request/builds/284#01811616-a528-466b-8f3a-e6ed268240c9

cargo hakari check failed?

**Comment by alissa-tung** [comment]
> https://buildkite.com/singularity-data/pull-request/builds/284#01811616-a528-466b-8f3a-e6ed268240c9
> 
> cargo hakari check failed?

thx, updated in [b005229](https://github.com/singularity-data/risingwave/pull/2917/commits/b005229a6bfabfcb5fb435071be96976f0893645)

**Comment by skyzh** [comment]
... and you'll need to `cargo sort`! We have a one-liner `./risedev check` to help you fix all such things.

File: src/frontend/test_runner/tests/plan_test_runner.rs [hunk 1]
```diff
@@ -12,6 +12,63 @@
...
+    }
+
+    run_tests(run_tests_args, tests, |test| {
+        let (test_case_name, file_name): (String, String) = test.clone().data;
+        let path = std::path::Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("tests")
+            .join("testdata")
+            .join(file_name);
+
+        let file_content = Runtime::new()
```
Comment (skyzh) [inline src/frontend/test_runner/tests/plan_test_runner.rs:64]: For file read, we can simply use `std::fs::read`

File: src/frontend/test_runner/tests/plan_test_runner.rs [hunk 2]
```diff
@@ -12,6 +12,63 @@
...
+            .unwrap()
+            .block_on(tokio::fs::read_to_string(path))
+            .unwrap();
+        Runtime::new()
```
Comment (skyzh) [inline src/frontend/test_runner/tests/plan_test_runner.rs:68]: `Runtime::new()` will spawn a multi-thread runtime, which is not expected in unit tests. Generally, #[tokio::test] only spawn one-thread runtime.

Would you please try using a single thread runtime?

Comment (skyzh) [reply src/frontend/test_runner/tests/plan_test_runner.rs:68]: https://docs.rs/tokio-macros/0.2.0/src/tokio_macros/lib.rs.html#209

```
                tokio::runtime::Builder::new()
                    .basic_scheduler()
                    .enable_all()
                    .build()
                    .unwrap()
                    .block_on(async { #body })
            }
```

**Comment by skyzh** [comment]
You may take https://github.com/risinglightdb/risinglight/pull/657 as a reference 🤣

Also sqlparser is using file-discovery based tests. If you are interested, you may also have a try in future PRs:

https://github.com/singularity-data/risingwave/tree/main/src/sqlparser/test_runner/tests

File: src/frontend/test_runner/tests/plan_test_runner.rs (hunk 1 above)
Comment (alissa-tung) [reply src/frontend/test_runner/tests/plan_test_runner.rs]: thx. fixed

**Review by skyzh — APPROVED** [review]
Good work, thanks!

**Pull request merged**